decidim-debates/lib/decidim/debates/test/factories.rb

```diff
@@ -18,20 +18,25 @@ def generate_localized_debate_title
       end_time { 1.day.from_now }
     end
 
-    trait :with_author do
+    trait :citizen_author do
       start_time { nil }
       end_time { nil }
       author do
         build(:user, organization: component.organization) if component
       end
     end
 
-    trait :with_user_group_author do
+    trait :official do
+      author { component.try(:organization) }
+    end
+
+    trait :user_group_author do
       author do
-        build(:user, organization: component.organization) if component
+        create(:user, organization: component.organization) if component
       end
+
       user_group do
-        build(:user_group, :verified, organization: component.organization, users: [author]) if component
+        create(:user_group, :verified, organization: component.organization, users: [author]) if component
```

Comment:
do we need to use `create` for some reason or we can keep using `build`?

Reply:
I had to use create because otherwise it didn't create the users wither, and since it wasn't persisted there were some tests that failed.